app/models/agents/post_agent.rb

```diff
@@ -102,6 +102,9 @@ def handle(data, payload = {})
         when 'json'
           headers['Content-Type'] = 'application/json; charset=utf-8'
           body = data.to_json
+        when 'xml'
+          headers['Content-Type'] = 'text/xml; charset=utf-8'
+          body = data.to_xml(root: (interpolated(payload)[:xml_root] || 'post'))
```

Comment:
@cantino well, I'm not a ruby programmer, but the `interpolated(payload)[...]` construct is used for `content_type` and `post_url` as well. Hence I think this way it is at least consistent.

Generally I think the agent's options are the primary source for `xml_root` et al